File: openapi-client/javascript-closure-angular/API/Client/Body.js
goog.provide('API.Client.body');

/**
 * Details for an Email
 * @record
 */
API.Client.Body = function() {}

/**
 * The Contact whom is the primary recipient of this email.
 * @type {!string}
 * @export
 */
API.Client.Body.prototype.to;

/**
 * The contact whom is the this email is from.
 * @type {!string}
 * @export
 */
API.Client.Body.prototype.from;

/**
 * The subject or title of the email
 * @type {!string}
 * @export
 */
API.Client.Body.prototype.subject;

/**
 * The main email contents.
 * @type {!string}
 * @export
 */
API.Client.Body.prototype.body;

File: openapi-client/scalaz/src/main/scala/org/openapitools/client/api/Body.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import Body._

case class Body (
  /* The Contact whom is the primary recipient of this email. */
  to: Option[String],
/* The contact whom is the this email is from. */
  from: Option[String],
/* The subject or title of the email */
  subject: Option[String],
/* The main email contents. */
  body: Option[String])

object Body {
  import DateTimeCodecs._

  implicit val BodyCodecJson: CodecJson[Body] = CodecJson.derive[Body]
  implicit val BodyDecoder: EntityDecoder[Body] = jsonOf[Body]
  implicit val BodyEncoder: EntityEncoder[Body] = jsonEncoderOf[Body]
}
